src/components/flat-table/flat-table-head/flat-table-head.component.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from "react";
+import StyledFlatTableHead from "./flat-table-head.style";
+import FlatTableRowHeader from "../flat-table-row-header";
+import { FlatTableRowProps } from "../flat-table-row";
+
+export interface FlatTableHeadProps {
+  /** Array of FlatTableRow. */
+  children: React.ReactNode;
+}
+
+const getRefs = (length: number) =>
+  Array.from({ length }, () => React.createRef<HTMLTableRowElement>());
+
+export const FlatTableHead = ({ children, ...rest }: FlatTableHeadProps) => {
+  const [rowHeights, setRowHeights] = useState<number[]>([]);
+  const refs = getRefs(React.Children.count(children));
+  let hasFlatTableRowHeader: boolean;
+
+  useEffect(() => {
+    if (React.Children.count(children) > 1) {
+      setRowHeights(
+        refs.map(
+          (ref) => ref.current?.clientHeight ?? /* istanbul ignore next */ 0
```

Comment:
This will only hit coverage as it will treat a 0 (which it will be in the unit tests unless I mock the height) as falsy and apply 0 again so I'd prefer to add a test to mock it or leave it as is